### ISSUE TITLE
no contrast in the title bar

### ISSUE THREAD

**Issue by 1ts** [nextcloud/server#47969]
Client Windows 10, 
Edge Version 128.0.2739.79 and Firefox 130.0

Server:
Nextcloud Version 30, latest.
Betriebssystem: Linux 5.15.0-119-generic x86_64
Version: 8.3.8

![14-09-_2024_11-33-06](https://github.com/user-attachments/assets/366dac47-eb79-4f5b-a70a-430aa9e3a7a4)

The contrast in the title bar is not available in dark design.
The icons remain black with a blue title bar

**Comment by melroy89** [comment]
So I seen this as well when using Nextcloud v30 (Hub 9) now.. Could it be maybe related to the selected Primary color ?

**Comment by melroy89** [comment]
Did you change the background color and/or primary color in your backend: `https://<domain>/settings/admin/theming`?

**Comment by melroy89** [comment]
Even if I set my background color darker, the menu icons changes to white (and the `#hex` code also becomes white). HOwever, after refreshing the icons becomes black again.

**Comment by Ramalama2** [comment]
Same here with firefox & safari (i don't use chrome)

**Comment by kyrosene** [comment]
same here with firefox. after updating to hub 9 the title bar contrast completely changed with all my wallpapers. the accent color detection seems to have been changed in some way too

**Comment by Phoenix616** [comment]
This fixes the default dark theme for me (at least the icons and I couldn't notice any other issues with a quick check on common apps):

```css
@media (prefers-color-scheme: dark) {
    :root {
        --color-background-plain-text: #ffffff;
    }
    .app-menu-icon__icon {
        filter: none!important;
    }
}
```
E.g. with the Custom CSS app: https://apps.nextcloud.com/apps/theming_customcss

**Comment by 9joshua** [comment]
Same issue here...
![Image](https://github.com/user-attachments/assets/6fb51cc9-71c8-44c3-84de-e12ff5b47166)

The odd thing is I have 2 instances both running Nextcloud Hub 9 (30.0.1). Only one instance has this issue - using the same browser.
Icons remain black regardless if a theme is used or which theme is applied.

**Comment by susnux** [comment]
The app icons are not related to your color schema used, but are only based on your background.

If you have selected a bright background then the icons will be black, if the background is dark then the icons will be white.
You can fine tune this by changing the background color to something different.

So the color @9joshua and @melroy89 use is too bright, `#1991D7` has too low contrast for white icons, thus black icons are used for accessibility. Required contrast would be >= 4.5

black icons | white icons
---|---
![Image](https://github.com/user-attachments/assets/808e5fe3-de23-4c5b-982e-feabde14bbfe)|![Image](https://github.com/user-attachments/assets/3062efe2-7f5a-4e35-90ed-68d76587a25b)

---

So for @1ts please share your configured background color. If you have manually configured a bright one then this is expected as the icon color is based on the background color.

BTW: If you choose an image the mean color is used, but you can fine tune the background color manually if you like.

**Comment by kyrosene** [comment]
did the logic for determining the color of the icons based on the background change at all within the last release? because many of my background images that used to result in white icons now use black icons

**Comment by 9joshua** [comment]
@susnux you are right, thank you. @kyrosene yes the same happened for me, what was background colour that met the threshold for white icons prior to HUB 9 no longer does and my UI now uses black icons. 

The main issue, and likely the same as the OPs, is that the icon colour no longer adjusts to personalised themes. For example, black icons based on the admin theme, remain black if a dark theme is chosen in the Personal theme preference...
![Image](https://github.com/user-attachments/assets/30832091-ef26-4138-8ae5-41f2472844ee)

Also, choosing a custom colour in the admin theme section doesn't seem to be remembered/applied at times. The icons will turn dark when choosing a lighter background colour, but revert to white when navigating to another page.

**Comment by susnux** [comment]
> The main issue, and likely the same as the OPs, is that the icon colour no longer adjusts to personalised themes. For example, black icons based on the admin theme, remain black if a dark theme is chosen in the Personal theme preference...

I think you are right this sounds like the real problem. I will put this on my todo, but just to warn you this might not be included in 30.0.2 but maybe 30.0.3.

**Comment by melroy89** [comment]
> Also, choosing a custom colour in the admin theme section doesn't seem to be remembered/applied at times. The icons will turn dark when choosing a lighter background colour, but revert to white when navigating to another page.

Yea correct, there is something definitely wrong in this whole theme color selection part of Nextcloud now. Even selecting a darker color seems to change the icons to white during that moment. After saving and reloading you will get black icons again. There seems to be different algorithms implemented maybe in JS.

**Comment by Gibibit** [comment]
I don't know if it's within the scope of this issue but a toggle to manually override this part of the color selection would be nice. This is the second time in a few months I've had to add custom CSS to fix the titlebar colors, and the logo is still dark now. Not a fan of the automatic contrast detection.

![Image](https://github.com/user-attachments/assets/eff59c5e-bd0f-4734-8297-1d4fcc8fbadb)

**Comment by susnux** [comment]
@Gibibit this looks like you have custom CSS as the menu and the app icons are sharing the same settings. So we can not help you there.

The color is based on the **background** color you configure in the theming settings (**NOT the primary color**).
If you upload an image the mean color of that image will be used as the background color.
See comments above how to tune this.

**Comment by susnux** [comment]
At all that suffers from this issue:

Please provide me this information:
1. Which theme do you use (light, dark, default, etc)
2. Which background do you use
3. Do you have admin theming background changed?
4. What is your admin theming background color?

If possible report the output of this:
```sh
occ config:app:get theming backgroundMime
occ config:app:get theming background_color
# Replace "admin" with your username below:
occ user:setting admin theming
```

**Comment by GretaD** [comment]
We are having this on mail too. Default theme selected, it happens when the browser is dark themed.

![Image](https://github.com/user-attachments/assets/b20b4553-fb91-4b29-8c0e-59dc84ff837d)

![Image](https://github.com/user-attachments/assets/3a2bcb68-5d79-46d5-b255-8ab8bf6c2953)
![Image](https://github.com/user-attachments/assets/faeab31f-033f-43a1-92e6-e5b0e0fb83cf)

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by 9joshua** [comment]
Just a post to prevent this from auto-closing. Important issue IMO.

**Comment by d-schiffner** [comment]
Same issue here
@susnux here the output of the commands:
```
#./occ status
  - installed: true
  - version: 30.0.4.1
  - versionstring: 30.0.4
  - edition:
  - maintenance: false
  - needsDbUpgrade: false
  - productname: Nextcloud
  - extendedSupport: false
```

```
#./occ config:app:get theming backgroundMime
<nothing>
```
```
#./occ config:app:get theming background_color
#FAFAFA
```

```
./occ user:setting <username> theming
  - theming:
    - enabled-themes: ["dark"]
  - settings:
    - display_name: <username>
```

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by susnux** [comment]
@Phoenix616 no the bot behaves like it should.
The ticket is still on "needs info" state because I requested some configuration output which was not provided (BTW @d-schiffner thank you for adding it :) ).

**Comment by bentuna** [comment]
Maybe this is related: When I choose a primary color, the button text color of the "choose" button (and all other buttons) seems to decide wrong about using dark or bright text color to match the button background color. 

But the hex value label above decides for the correct text color.

<img width="343" alt="Image" src="https://github.com/user-attachments/assets/19099b0b-7694-4e48-b8cb-ee161c895974" />

**Comment by isdnfan** [comment]
For me in NC31 most apps look good - even Office is OK (but differs) -pdf viewer has has black icons on dark theme

![Image](https://github.com/user-attachments/assets/cc9bd077-e826-4c73-bd6a-705713ed1e8c)

**Comment by susnux** [comment]
@isdnfan There are no icons on the PDF viewer thats a black backdrop as the viewer overlays the whole page.

**Comment by susnux** [comment]
@bentuna the color of the button will only change once you picked a color and the updated theme is downloaded from server to apply.

**Comment by isdnfan** [comment]
> [@isdnfan](https://github.com/isdnfan) There are no icons on the PDF viewer thats a black backdrop as the viewer overlays the whole page.

I see, thank you for clarification. It still feels wrong/bad. IMHO the reader should not overlay the title bar or it should completely cover the buttons (I would prefer 1st for the sake of consistency but 2nd is OK as well.)

**Comment by bentuna** [comment]
> [@bentuna](https://github.com/bentuna) the color of the button will only change once you picked a color and the updated theme is downloaded from server to apply.

To make it intuitive, the text color should change immediately, like it does for the Button marked green in my screenshot.

But also, the text color does not change later. It stays wrong (black text still on too dark background color). The algorithm which decides when to use white and when to use black Text seems to be wrong. So there seems to be another algorithm for that decision in the green marked button than in the red marked button. We should use the algorithm from the green marked button for the whole Nextcloud.

**Comment by bentuna** [comment]
> > [@isdnfan](https://github.com/isdnfan) There are no icons on the PDF viewer thats a black backdrop as the viewer overlays the whole page.
> 
> I see, thank you for clarification. It still feels wrong/bad. IMHO the reader should not overlay the title bar or it should completely cover the buttons (I would prefer 1st for the sake of consistency but 2nd is OK as well.)

Agreed. I would prefer the second option (100% opacity for the black overlay) for the sake of efficient space usage and maximum clarity (no slight shine-through of the app symbols).

**Comment by joshtrichards** [comment]
Related: #35300

**Comment by susnux** [comment]
Still a bit unsure what to do here, this is basically a configuration issue if you use the two colored default background but the admin setup a different background color.

Easiest fix is the admin fixing their config:
- Remove the admin background color / reset it if you have the default background in the admin settings.

Otherwise for us its probably disable the admin color if the background image is still the default one.
